Assets/Scripts/Managers/BattleQueue.cs

```diff
@@ -280,11 +410,426 @@ public List<ActionClass> GetList()
     }
 
 
+
+    // Class for wrappers to be contained 
+    public class WrapperArray
+    {
+        private List<Wrapper> wrappers = new List<Wrapper>();
+
+        public WrapperArray()
+        {
+            // nothing here. 
+        }
+
+        // is called when roundStart is true. 
+        // transforms all enemy actions into wrappers. 
+        // is automatically sorted because iterating through a sorted array.
+
+        // peep the redundancy is referncing and calling
+        public void EnemyActionsTransformation()
+        {
+            for (int i = 0; i < BattleQueueInstance.actionQueue.GetList().Count; i++) // !!! why was BattleQueueInstance necessary here; object reference for non static field? 
+            {
+                ActionClass curAction = BattleQueueInstance.actionQueue.GetList()[i];
+                Wrapper wrapper = new Wrapper(curAction, true); // funnily enough this won't work in just C; works here because new is dynamic allocation.
+                wrappers.Add(wrapper);
+            }
+            // DisplayWrapperArray();
+
+            BattleQueue.BattleQueueInstance.roundStart = false;
+
+            // Deactivate the actionQueue;
+            // this is flawed at present and I see no point in removing this as it is already implemented; reimplementing or refactoring this functionality should come later
+            // for reference notice the implementaion of methods inside *this* class assume implementation of that.
+            /*            while (BattleQueueInstance.actionQueue.GetList().Count != 0)
+                        {
+                            BattleQueueInstance.actionQueue.GetList().Remove(BattleQueueInstance.actionQueue.GetList()[0]);
+                        }
+                        // alternatively
+                        BattleQueueInstance.actionQueue.GetList().Clear();*/
+        }
```

Comment:
Remove these comments please, imo they detract from code understanding instead of enhance.